### ISSUE TITLE
Firefox resizing

### ISSUE THREAD

**Issue by DanielTate** [extralagence/extra.slider#2]
Re-sizing doesn't scale in Firefox it crops the images on the slider, this is reproducible on the current website.

Looking into this it looks like a fault in the getDimensions function, I am attempting to fix now I have resolved the width issue now working on height, it ultimately needs to be refactored, the getDimensions funciton is being called 4 times during setup in each call its looping through all the list items to check their height, this seems very expensive and maybe can be optimized. I will make a pull request if I find a comprehensive solution.

EDIT:

So my solution was to add to the get Dimensions function before the return:

```
            if(type === 'width'){
                if (item.parent().outerWidth() < max) {
                    max = item.parent().outerWidth();
                }
            }

            if($(window).width() >= 320 && type === 'height'){
                max = 'auto';
            }
```

Obviously this will vary case by case and it works for the width by checking the parents restrictions and keeping them in mind when applying width, the height is a hack and works but would not help if you want things to be responsive based on the height of the element.

The function really needs to be re implemented with the item loop abstracted, I think this would lead to a faster runtime and less issues on resizes.

**Comment by jbjanot** [comment]
Hi Daniel,

Thanks a lot for your feedback. I am going to look at your proposed solution, it seems great.
 I have some other improvements that needs to be added, so I guess your message is a good opportunity to spend some time on this.

See you !

**Comment by jbjanot** [comment]
Hi ! I just pushed a new version. Nothing really revolutionary, but by the way I managed to identify the problem with Firefox (it was only a missing max-width).
I still have to take a deeper look in performance improvements, but that would be for another update.

The demos now have js examples, and the website documentation is up-to-date : 
http://slider.extralagence.com/documentation/
http://slider.extralagence.com/demos/

Any feedback is much appreciated !
See you.
